src/components/Layout/Layout.js

```diff
@@ -0,0 +1,33 @@
+import { Fragment, useState, useEffect } from "react";
+import { getUser } from "../../api/api";
+import Footer from "../Footer/Footer";
+import NavBar from "../NavBar/NavBar";
+
+const Layout = ({ children }) => {
+  const [userProfile, setUserProfile] = useState({
+    email: null,
+    profileImageSource: null,
+  });
+
+  const getApiData = async () => {
+    try {
+      const userData = await getUser();
+      setUserProfile(userData);
+    } catch (error) {
+      console.error("Error fetching user data:", error);
+    }
+  };
+
+  useEffect(() => {
+    getApiData();
+  }, []);
+  return (
+    <Fragment>
+      <NavBar userProfile={userProfile} />
+      <main>{children}</main>
+      <Footer />
+    </Fragment>
+  );
```

Comment:
P3:
이런 경우 fragment를 써야합니다. 잘사용하셨네요!
참고로 아래처럼도 가능합니다.

```suggestion
  return (
    <>
      <NavBar userProfile={userProfile} />
      <main>{children}</main>
      <Footer />
    </>
  );
```